src/etc/dhcp/dhclient-exit-hooks.d/98-vyos-static-routes-dhclient-hook

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#
+# Copyright VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+DHCP_HOOK_IFLIST="/tmp/static_dhcp_interfaces"
+
+# Only run if there are static routes with dhcp-interface configured
+if ! { [ -f $DHCP_HOOK_IFLIST ] && grep -qw $interface $DHCP_HOOK_IFLIST; }; then
+    return 0
+fi
+
+# Re-generate the config on the following events:
+# - BOUND: always re-generate
+# - RENEW: re-generate if the IP address changed
+# - REBIND: re-generate if the IP address changed
+# - EXPIRE: always re-generate (route should be removed)
+# - RELEASE: always re-generate (route should be removed)
+if [ "$reason" == "RENEW" ] || [ "$reason" == "REBIND" ]; then
+    if [ "$old_ip_address" == "$new_ip_address" ]; then
+        return 0
+    fi
+elif [ "$reason" != "BOUND" ] && [ "$reason" != "EXPIRE" ] && [ "$reason" != "RELEASE" ]; then
+    return 0
+fi
+
+# Best effort wait for any active commit to finish
+sudo python3 - <<PYEND
+from vyos.utils.commit import wait_for_commit_lock
+
+if __name__ == '__main__':
+    wait_for_commit_lock()
+    exit(0)
+PYEND
+
+# Now re-generate the static routes config
+sudo /usr/libexec/vyos/conf_mode/protocols_static.py
```

Comment:
What will happen if an instance is deployed in a cloud?
During the first boot, it gets an IP address by DHCP before the full config is loaded. Then, after the address is obtained via DHCP, based on the metadata, it gets the other router's config.
Not sure what will be with `wait_for_commit_lock()` or `sudo /usr/libexec/vyos/conf_mode/protocols_static.py` in this case. Will this hook fail?

Good example for the working hook https://github.com/vyos/vyos-1x/blob/current/src/etc/dhcp/dhclient-enter-hooks.d/05-vyos-mtureplace